src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -0,0 +1,614 @@
+    bool configUpdated = false;
+    bool wifiSleepMode = cfg_.wifi_sleep().read(fd);
+    bool hubEnabled = cfg_.hub_config().enabled().read(fd);
+    uint16_t hubPort = cfg_.hub_config().port().read(fd);
+    String hubServiceName = String(cfg_.hub_config().service_name().read(fd).c_str());
+    const TcpClientConfig<TcpClientDefaultParams> &link_config =
+        cfg_.link_config();
+    SocketClientParams::SearchMode tcpClientMode =
+        (SocketClientParams::SearchMode)link_config.search_mode().read(fd);
+    String tcpmDNSServiceName =
+        String(link_config.auto_address().service_name().read(fd).c_str());
+    String tcpAutoHostName =
+        String(link_config.auto_address().host_name().read(fd).c_str());
+    String tcpManualHostName =
+        String(link_config.manual_address().ip_address().read(fd).c_str());
+    uint16_t tcpManualPort = link_config.manual_address().port().read(fd);
+
+    // if this is the first time this node is loading its config we can simply
+    // accept all config as is and start the WiFi system.
+    if (initial_load)
+    {
+        configUpdated = true;
+        wifiSleepMode_ = wifiSleepMode;
+        hubEnabled_ = hubEnabled;
+        hubPort_ = hubPort;
+        hubServiceName_ = hubServiceName;
+        tcpClientMode_ = tcpClientMode;
+        tcpmDNSServiceName_ = tcpmDNSServiceName;
+        tcpAutoHostName_ = tcpAutoHostName;
+        tcpManualHostName_ = tcpManualHostName;
+        tcpManualPort_ = tcpManualPort;
+    }
+    else
+    {
+        // check each parameter and update the cache if it is different
+        VERIFY_CACHED_VALUE(wifiSleepMode_, wifiSleepMode)
+        VERIFY_CACHED_VALUE(hubEnabled_, hubEnabled)
+        VERIFY_CACHED_VALUE(hubPort_, hubPort)
+        VERIFY_CACHED_VALUE(hubServiceName_, hubServiceName)
+        VERIFY_CACHED_VALUE(tcpClientMode_, tcpClientMode)
+        VERIFY_CACHED_VALUE(tcpmDNSServiceName_, tcpmDNSServiceName)
+        VERIFY_CACHED_VALUE(tcpAutoHostName_, tcpAutoHostName)
+        VERIFY_CACHED_VALUE(tcpManualHostName_, tcpManualHostName)
+        VERIFY_CACHED_VALUE(tcpManualPort_, tcpManualPort)
+    }
+
+    // verify settings provided to make sure they are valid
+    if (tcpClientMode_ != SocketClientParams::SearchMode::AUTO_ONLY)
+    {
+        // if the manual hostname is not provided (or is blank) or we do not
+        // have a valid manual port force settings to AUTO_ONLY.
+        if (!tcpManualHostName_.length() || tcpManualPort_ == 0)
+        {
+            LOG(WARNING, "Missing manual hub hostname or port, "
+                "switching to AUTO_ONLY mode");
+            tcpClientMode_ = SocketClientParams::SearchMode::AUTO_ONLY;
+            // record the updated settings.
+            link_config.search_mode().write(fd, tcpClientMode_);
+        }
+    }
+
+    // if this is the first time we are loaded start the WiFi system.
+    if (initial_load)
+    {
+        startWiFiSystem();
```

Comment:
This is a blocking call. That breaks the contract of this API.

Reply:
ok, let me rethink this one.